packages/components/src/Disclosure/src/Disclosure.tsx

```diff
@@ -0,0 +1,97 @@
+import { useStyledSystem, type StyledComponentProps } from "@hopper-ui/styled-system";
+import { forwardRef, useState, type ForwardedRef } from "react";
+import { composeRenderProps, Disclosure as RACDisclosure, useContextProps, type DisclosureProps as RACDisclosureProps } from "react-aria-components";
+
+import { composeClassnameRenderProps, cssModule, SlotProvider } from "../../utils/index.ts";
+
+import { DisclosureContext } from "./DisclosureContext.ts";
+import { DisclosureHeaderContext } from "./DisclosureHeaderContext.ts";
+import { DisclosurePanelContext } from "./DisclosurePanelContext.ts";
+import { InternalDisclosureContext } from "./InternalDisclosureContext.ts";
+
+import styles from "./Disclosure.module.css";
+
+export const GlobalDisclosureCssSelector = "hop-Disclosure";
+
+export type DisclosureVariant = "standalone" | "inline";
+
+export interface DisclosureProps extends StyledComponentProps<RACDisclosureProps> {
+    variant?: DisclosureVariant;
+}
+
+function Disclosure(props: DisclosureProps, ref: ForwardedRef<HTMLDivElement>) {
+    [props, ref] = useContextProps(props, ref, DisclosureContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children: childrenProp,
+        style: styleProp,
+        variant = "standalone",
+        ...otherProps
+    } = ownProps;
+
+    const classNames = composeClassnameRenderProps(
+        className,
+        GlobalDisclosureCssSelector,
+        cssModule(
+            styles,
+            "hop-Disclosure",
+            variant
+        ),
+        stylingProps.className
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...stylingProps.style,
+            ...prev
+        };
+    });
+
+    const children = composeRenderProps(childrenProp, prev => {
+        return prev;
+    });
+
+    const [hasHeader, setHasHeader] = useState(false);
```

Comment:
@victortrinh2 maybe with `:has`

Reply:
there is the useSlot hook for this

Reply:
oh I forgot about this hook, thanks!. as for :has in css, yes I used that at first, but it can only be used inside the Disclosure css and not the disclosure panel css because of the way css modules work.